javascript/code-challenges/challenges-10.test.js

```diff
@@ -0,0 +1,347 @@
+CHALLENGE 2
+
+Write a function named findMax that takes in a matrix of positive numbers and returns the number with the highest value.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 6],
+  [23, 5, 5]
+]
+
+return: 23
+------------------------------------------------------------------------------------------------ */
+const findMax = (matrix) => {
+  let max = 0;
+  matrix.forEach(element => {
+    element.forEach(value => {
+      if (value > max){
+        max = value;
+      }
+    })
+  });
+  return max;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named totalSum that takes in a matrix of numbers and returns the totalSum of all the numbers.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 1],
+  [2, 5, 5]
+]
+
+return: 35
+------------------------------------------------------------------------------------------------ */
+const totalSum = (matrix) => {
+  let a = 0;
+   matrix.forEach(value => {
+
+  a = a + value;
+  })
+  return a;
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+You friend Pat has a chain of stores around the greater Seattle area. He specializes in selling salmon cookies. Pat has data for the hourly sales of cookies per hour for each store. He wants to create an array of the total number of cookies sold per hour for all of his stores combined.
+
+Write a function named grandTotal that adds up the cookies sales for each hour of operation for all of the stores combined. For example, the first element in the hourlySales array should be the sum of the cookies sold in the 9:00 a.m. hour at all five stores combined.
+
+For this example, the total at 9:00 a.m. is 17 + 26 + 7 + 5 + 33, or 88 total cookies.
+
+Return the array of the total number of cookies sold per hour for all of the stores combined.
+------------------------------------------------------------------------------------------------ */
+
+const hoursOpen = ['9 a.m.', '10 a.m.', '11 a.m.', '12 p.m.', '1 p.m.', '2 p.m.', '3 p.m.', '4 p.m.', '5 p.m.', '6 p.m.', '7 p.m.', '8 p.m.'];
+
+const firstPike = [17, 18, 23, 24, 24, 12, 13, 27, 30, 20, 24, 18];
+const seaTac = [26, 5, 5, 59, 23, 39, 38, 20, 30, 7, 59, 43];
+const seattleCenter = [7, 14, 19, 22, 15, 4, 23, 27, 28, 23, 1, 29];
+const capHill = [5, 85, 58, 51, 50, 13, 33, 32, 47, 94, 31, 62];
+const alkiBeach = [33, 31, 147, 130, 27, 93, 38, 126, 141, 63, 46, 17];
+
+const cookieStores = [firstPike, seaTac, seattleCenter, capHill, alkiBeach];
+
+const grandTotal = (stores) => {
+   
+      let totalSum = 0;
+      let cookiesSoldEachHourArray = [];
+      for (let i = 0; i < this.hoursOpen.length - 1; i++) { 
+              let storesperhour = firstPike[i] + seaTac[i] + seattleCenter[i] + capHill[i] + alkiBeach[i];
+      cookiesSoldEachHourArray.push(storesperhour); 
```

Comment:
try automating this. You're manually typing out each store. What if we added more stores to the cookieStores array? You'd have to come back in and re-write your function. Where at all possible you want your functions to be scalable and non-specific. If we added a 100 new store locations you'd still want your function to work. But, good news there's already and array of all the stores, so you can simply automate iterating through that array.